### ISSUE TITLE
[pickers] Handle time selection correctly regardless of date being set

### ISSUE THREAD

**Issue by LukasTy** [mui/mui-x#6417]
When using `DateTimeField` and changing time—it does not correctly cycle through am/pm time until the whole date is set.
Only after setting a valid date it works as expected by changing am/pm when changing time numbers.

Same applies to `TimeField`—when changing `hours` without `minute` being set—the component does not cycle through `am/pm.

https://user-images.githubusercontent.com/4941090/194510029-d95a3b7a-2282-4aff-ad67-965646fbc8d9.mov

**Comment by michelengelen** [comment]
This is fixed by the field rewrite. The current section-editing model (`adjustLetterSection` in the field root props) derives the AM/PM options independently of the other sections' values, it always resolves them against the current date, not the field's in-progress value. So pressing ArrowUp/ArrowDown on the meridiem section now cycles AM/PM immediately, whether or not hours or minutes have a value yet.

Closing as completed. If a specific adapter or locale still shows the old coupling, please open a new issue naming it.
